.github/workflows/test_iasworld_data.yaml

```diff
@@ -0,0 +1,141 @@
+# Department's Spark data extraction process dispatches it via API once per
+# day after finishing the daily ingest of iasWorld data. For more detail, see:
+#
+# https://github.com/ccao-data/service-spark-iasworld
+on:
+  workflow_dispatch:
+    inputs:
+      select:
+        description: >
+          Optional space-separated list of tests to run (defaults to all
+          iasWorld data tests)
+        required: false
+        type: string
+      selector:
+        description: >
+          Optional dbt selector representing tests to run (takes precedence
+          over the above list of tests if both are present)
+        required: false
+        type: string
+      upload_test_results:
+        description: Upload test results to S3
+        required: false
+        default: false
+        type: boolean
+
+env:
+  PYTHONUNBUFFERED: "1"
+  UV_SYSTEM_PYTHON: 1
+
+jobs:
+  test-iasworld-data:
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint
+    # so that we can authenticate with AWS
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Validate and parse input variables
+        id: parse-inputs
+        run: |
+          # Default to no select option, which will fall back to the default
+          # behavior of the underlying Python script
+          SELECT_OPTION=""
+          if [[ -n "$SELECTOR" ]]; then
+            SELECT_OPTION="--selector $SELECTOR"
+          elif [[ -n "$SELECT" ]]; then
+            SELECT_OPTION="--select $SELECT"
+          fi
+          echo "Setting select option to '$SELECT_OPTION'"
+          echo "select-option=$SELECT_OPTION" >> "$GITHUB_OUTPUT"
+        shell: bash
+        env:
+          SELECT: ${{ inputs.select }}
+          SELECTOR: ${{ inputs.selector }}
+
+      - name: Setup dbt
+        uses: ./.github/actions/setup_dbt
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
+
+      - name: Restore dbt state cache
+        id: cache
+        uses: ./.github/actions/restore_dbt_cache
+        with:
+          path: ${{ env.PROJECT_DIR }}/${{ env.STATE_DIR }}
+          key: ${{ env.CACHE_KEY }}
+
+      - name: Install Python dependencies
+        working-directory: ${{ env.PROJECT_DIR }}
+        shell: bash
+        run: uv pip install ".[dbt_tests]"
+
+      - name: Run tests
+        run: |
+          DEFER_OPTION=""
+          if [[ "$CACHE_HIT" == 'true' ]]; then
+            DEFER_OPTION="--defer --state $STATE_DIR"
+          fi
+          # shellcheck disable=SC2086
+          python scripts/run_iasworld_data_tests.py \
+            --target "$TARGET" \
+            --output-dir ./qc_test_results/ \
+            $SELECT_OPTION \
+            $SKIP_ARTIFACTS_OPTION \
+            $DEFER_OPTION
+        working-directory: ${{ env.PROJECT_DIR }}
+        shell: bash
+        env:
```

Comment:
**question (non-blocking):** Why pass these in as env vars rather than just interpolating them directly into the bash?

Reply:
Good question! There's no functional reason, I just felt like it was clearer in this particular case. For `USER`, `GIT_SHA`, `GIT_REF`, and `GIT_AUTHOR`, the script expects these environment variables to be set, so we need to convert the GitHub variables to env vars anyway. For `SELECT_OPTION`, `CACHE_HIT`, and `SKIP_ARTIFACTS_OPTION`, I felt like the GitHub variable references were a bit long/esoteric, and it feels easier to me to read them as env vars. Definitely a weak preference on my part, I'm open to changing any of these if you think a the GitHub variables would be clearer.